### PR TITLE
Refactor and add proper reducer tests

### PR DESCRIPTION
**Pull request by vsaarinen**
The reducer tests have been a mess. Split them up (similarly to what was done with sagas and their tests), bring them up to date and add missing or pending tests for all reducers.

TODO:
- [x] Activities
- [x] Branches
- [x] Comments
- [x] Commits
- [x] Deployments
- [x] Errors
- [x] Modal
- [x] Previews
- [x] Projects
- [x] Requests
- [x] Selected
- [x] Streaming
- [x] User
- [x] Rebase onto master once #263 is merged

### CODE REVIEW

**Comment by dennari** [comment]
Number of specs increased by 121%. Superb.

File: src/js/modules/activities/reducer.ts [hunk 1]
```diff
@@ -3,16 +3,16 @@ import { Reducer } from 'redux';
...
 
-const initialState: t.ActivityState = {};
+const initialState: ActivityState = {};
 
-const reducer: Reducer<t.ActivityState> = (state: t.ActivityState = initialState, action: any) => {
+const reducer: Reducer<ActivityState> = (state: ActivityState = initialState, action: any) => {
   switch (action.type) {
     case STORE_ACTIVITIES:
-      const activities = (action as t.StoreActivitiesAction).entities;
+      const activities = (action as StoreActivitiesAction).entities;
```
Comment (dennari) [inline src/js/modules/activities/reducer.ts]: Isn't there some way to make TypeScript automatically infer the correct type here?

Comment (vsaarinen) [inline src/js/modules/activities/reducer.ts]: Yep, but it requires quite a bit of work: we need to make sure all the actions we use are properly typed (i.e. don't extend Action, which is defined as `{ type: string; }`), then we need to gather all the our Action types into one type, which should then be used as the action type for all our reducers. I could do that later in a separate PR—the recent work related to getting rid of inherited Action types was preparation for it.

Comment (vsaarinen) [inline src/js/modules/activities/reducer.ts]: We also need to add types for actions from external Redux libraries to this super Action type.

Comment (dennari) [inline src/js/modules/activities/reducer.ts]: Check

**Comment by dennari** [comment]
:+1:

**Pull request merged**